test/detail/test_function.cpp

```diff
@@ -546,6 +535,19 @@ BOOST_FIXTURE_TEST_CASE( triggering_an_expectation_returns_the_set_auto_ptr_valu
         BOOST_CHECK( ! f().get() );
         CHECK_CALLS( 2 );
     }
+    {
+        mock::detail::function< std::auto_ptr< int >() > f;
```

Comment:
Should we bother maintaining `auto_ptr` support?

Reply:
Good point. 150 lines less code to maintain